### PR TITLE
Searchbox

### PR DESCRIPTION
**Pull request by RachelCohen1**
searchbar component

### CODE REVIEW

**Comment by RachelCohen1** [comment]
searchbox and searchbox with autocomplete option

File: components/search-bar/_search-bar.scss [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+.action-button{
+    cursor:pointer;
+    cursor: pointer;
+    position: relative;
+    float: right;
+    right: 11px;
+}
+
```
Comment (islavi) [inline components/search-bar/_search-bar.scss]: Need to wrap all the styles below parent class: search-bar-container

In case the user type text that exceed the input width the icon will be covered with text.
Need to add specific padding-right: 30px; to: search-bar-input class.

The style: +.active{
 +  .sdc-input .sdc-input__input{
 +    border-color: $blue;
 +  }
 +}
is relevant for inputs, remove it from search scss

File: src/angular/searchbox/search-bar.component.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+})
+export class SearchBarComponent {
+
+    @Input() public placeholder: string;
+    @Input() public label: string;
+    @Input() public searchQuery: string;
+    @Input() public debounceTime: number = 200;
+    @Output() public searchChanged: EventEmitter<any> = new EventEmitter<any>();
+    @Output() public searchButtonClicked: EventEmitter<string> = new EventEmitter<string>();
+
```
Comment (islavi) [inline src/angular/searchbox/search-bar.component.ts]: @lizronen @michalweinsh @einavk 
Liz, Michal and Einav please tell what do you think

It is not clear why we have: searchChanged and searchButtonClicked.

It looks like these are 2 different components:

1. **Search bar**: 
- perform action on click. 
- magnify change to "X" after clicking magnify (and not while typing).
- has event searchButtonClicked

2. **Filter bar** 
- perform action while writing. 
- magnify change to "X" after typing value.
- has event searchChanged (2 ways binding)

File: stories/ng2-component-lab/checklist.component.exp.ts [hunk 3]
```diff
@@ -147,7 +147,7 @@ export default experimentOn('Checklist')
         `
     },
     {
-        id: 'multiLevelsChecklist',
```
Comment (islavi) [inline stories/ng2-component-lab/checklist.component.exp.ts:150]: why did you change this file ? 
It is not relevant to search

File: stories/ng2-component-lab/components/search-bar-example.component.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
+/**
+ * Created by rc2122 on 11/15/2017.
+ */
+import {Component} from "@angular/core";
+
+@Component({
+    selector: "search-bar-example",
+    template: `
+        <sdc-search-bar placeholder="search text"
```
Comment (islavi) [inline stories/ng2-component-lab/components/search-bar-example.component.ts]: This should be experiment by itself, because in experiment user see only: <search-bar-example></search-bar-example>
and dose not understand how to write search component.

File: stories/ng2-component-lab/input.component.exp.ts [hunk 5]
```diff
@@ -3,6 +3,10 @@
  */
 import {experimentOn} from '@islavi/ng2-component-lab';
 
+const valueChange = (value: any): void => {
```
Comment (islavi) [inline stories/ng2-component-lab/input.component.exp.ts:6]: why did you change this file in search component?
If we need to fix the input please add another branch.

File: stories/ng2-component-lab/search-bar.component.exp.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+                private list: string[] = ['apple', 'banana', 'orange', 'peach'];
+                private searchText: string = '';
+                private onSearchTextChange = (value: string): void => {
+                    this.searchText = value;
+                };
+            }
+        </pre>`,
+        showSource: true,
+        template: `
+        <search-bar-example></search-bar-example>
```
Comment (islavi) [inline stories/ng2-component-lab/search-bar.component.exp.ts]: See my remark above. 
It is not clear to the user how to build the search component.
Is should be like this:
<sdc-search-bar placeholder="search text"
                        label="search box example:"
                        [debounceTime]="1000"
                        [searchQuery]="searchText"
                        (searchChanged)="onSearchTextChange($event)">

**Review by islavi — CHANGES_REQUESTED** [review]
I did not review the search with auto complete yet.
Can you split it to different branch ?
